src/clients/c/tb_client/context.zig

```diff
@@ -114,16 +118,24 @@ pub fn ContextType(
         thread: std.Thread,
 
         pub fn init(
-            allocator: std.mem.Allocator,
+            root_allocator: std.mem.Allocator,
             cluster_id: u128,
             addresses: []const u8,
             completion_ctx: usize,
             completion_fn: tb_completion_t,
         ) Error!*Context {
-            var context = try allocator.create(Context);
-            errdefer allocator.destroy(context);
+            var context = try root_allocator.create(Context);
+            errdefer root_allocator.destroy(context);
+
+            // Wrap the root allocator - presumed to be heap.c_allocator - in a GPA to keep maximum
+            // compatibility while gaining the extra safety. As a library, libtbclient is running
+            // inside another process's address space.
+            context.gpa = GPA{
+                .backing_allocator = root_allocator,
+            };
+            const allocator = context.gpa.allocator();
+            errdefer assert(context.gpa.deinit() == .ok);
```

Comment:
We can safely _move_ the GPA since we don't keep live references to `gpa.allocator()`.
So we don't need to allocate `context` differently from `root_allocator`.

```suggestion
            const context: *Context = context: {
                // Wrap the root allocator - presumed to be heap.c_allocator - in a GPA to keep maximum
                // compatibility while gaining the extra safety. As a library, libtbclient is running
                // inside another process's address space.
                var gpa = GPA{
                    .backing_allocator = root_allocator,
                };
                errdefer assert(gpa.deinit() == .ok);

                const context = try gpa.allocator().create(Context);
                // Moving the GPA is safe, since we don't have any live reference to `allocator`.
                context.gpa = gpa;
                break :context context;
            };
            errdefer {
                var gpa: GPA = context.gpa;
                gpa.allocator().destroy(context);
            }
            const allocator = context.gpa.allocator();
```

Reply:
Ah yes, wrapping it inside a block is clever! I had initially done it by moving the allocator externally, and the code wasn't easy to follow.